### PR TITLE
add spam prevention antehandler

### LINKED ISSUE #2246: Gov Spam Prevention Mechanism
## Summary
A spike in spam gov proposals has necessitated a temporary mitigation until the Hub can upgrade to v0.47 of the SDK which will include enforceable spam prevention via the addition of a gov param for [initial proposal submission deposit.](https://github.com/cosmos/cosmos-sdk/pull/12771). 

## Problem Definition

At present it costs next to nothing to execute a spam attack via governance on the Hub. While the only nuisance to the chain it causes is driving a sharp increase in the gov proposal index, most of these text proposals include malicious links that have the potential to harm users who might be unaware.

## Proposal

Create a new ante decorator that filters messages for `MsgSubmitProposal`, and returns an error if the minimum deposit threshold is not met. This `MinInitialDeposit` can be a hard amount, or a percentage of the `MinDeposit`. Current `MinDeposit` is `250000000uatom`. Something like 10% would put the price to submit a proposal between 300-350$ at the time of writing.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by MSalopek**
This PR adds governance spam prevention antehandler and minimal testing.

Earlier, a github member opened a PR which they closed after receiving feedback:
- https://github.com/cosmos/gaia/pull/2251

The changes were modelled after:
- https://github.com/CosmosContracts/juno/pull/394

Closes: #2246

### CODE REVIEW

**Review by sainoe — APPROVED** [review]
Nice work @MSalopek! LGTM

**Comment by MSalopek** [comment]
~Let's not merge this yet, global fee e2e tests are not working. I'm trying to figure out why~

Done, tests are up to date.

**Review by mpoke — APPROVED** [review]

File: ante/gov_ante.go
Comment (tbruyelle) [inline ante/gov_ante.go:64]: @MSalopek Sorry to ask after the merge, but couldn't this code be simplified by calling `MsgExec.GetMessages()` ?

**Pull request merged**
